File: core/src/com/broll/gainea/server/core/cards/impl/play/C_TeleportShip.java
package com.broll.gainea.server.core.cards.impl.play;

import com.broll.gainea.server.core.cards.TeleportCard;
import com.broll.gainea.server.core.map.Location;

import java.util.List;

public class C_TeleportShip extends TeleportCard {
    public C_TeleportShip() {
        super(31, "Seeleute", "Bewegt eine Truppe zu einem beliebigen Schiff auf der gleichen Karte");
    }

    @Override
    public List<? extends Location> getTeleportTargets(Location from) {
        return from.getContainer().getExpansion().getAllShips();
    }
}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_TeleportShip.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.cards.TeleportCard
import com.broll.gainea.server.core.map.Location

class C_TeleportShip : TeleportCard(31, "Seeleute", "Bewegt eine Truppe zu einem beliebigen Schiff auf der gleichen Karte") {
    override fun getTeleportTargets(from: Location) =
            from.container.expansion.allShips

}
